sklearn/multioutput.py

```diff
@@ -474,6 +474,40 @@ def predict_proba(self, X):
         results = [estimator.predict_proba(X) for estimator in self.estimators_]
         return results
 
+    def _check_decision_function(self):
+        if hasattr(self, "estimators_"):
+            # raise an AttributeError if `decision_function` does not exist for
+            # each estimator
+            [getattr(est, "decision_function") for est in self.estimators_]
+            return True
+        # raise an AttributeError if `decision_function` does not exist for the
+        # unfitted estimator
+        getattr(self.estimator, "decision_function")
+        return True
+
+    @available_if(_check_decision_function)
+    def decision_function(self, X):
+        """Return confidence scores for each class of each output.
+
+        This method will raise a ``AttributeError`` if any of the
+        estimators do not have ``decision_function``.
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix???? of shape (n_samples, n_features)
```

Comment:
Why the `????`?

Reply:
The check will be delegated to the underlying estimator so we can mention `{array-like, sparse matrix}`.